Sources/Sentry/SentryPerformanceTracker.m

```diff
@@ -0,0 +1,182 @@
+#import "SentryPerformanceTracker.h"
+#import "SentryLog.h"
+#import "SentrySDK+Private.h"
+#import "SentrySpan.h"
+#import "SentrySpanProtocol.h"
+#import "SentryTracer.h"
+
+static NSString *const SENTRY_PERFORMANCE_TRACKER_SPANS = @"SENTRY_PERFORMANCE_TRACKER_SPANS";
+static NSString *const SENTRY_PERFORMANCE_TRACKER_ACTIVE_STACK
+    = @"SENTRY_PERFORMANCE_TRACKER_ACTIVE_STACK";
+
+/*
+ * Auxiliary class to store tracking information.
+ */
+@interface SentrySpanTracker : NSObject
+
+@property (nonatomic) id<SentrySpan> span;
+@property (nonatomic) BOOL finished;
+@property (nonatomic) SentrySpanStatus finishedStatus;
+@property (nonatomic, strong) NSMutableDictionary<NSString *, SentrySpanTracker *> *children;
```

Comment:
`m`: `SentrySpanId` implements `hash` and `isEqual`, maybe it makes sense to implement `NSCopying` so we can do this instead?
```suggestion
@property (nonatomic, strong) NSMutableDictionary<SentrySpanId *, SentrySpanTracker *> *children;
```